src/Microsoft.AspNet.Mvc.ViewFeatures/ViewExecutor.cs

```diff
@@ -36,7 +36,20 @@ public static class ViewExecutor
         {
             var response = actionContext.HttpContext.Response;
 
-            contentType = contentType ?? DefaultContentType;
+            if (contentType == null)
+            {
+                contentType = DefaultContentType;
+
+                if (!string.IsNullOrEmpty(response.ContentType))
+                {
+                    MediaTypeHeaderValue parsedContentType;
+                    if (MediaTypeHeaderValue.TryParse(response.ContentType, out parsedContentType))
+                    {
+                        contentType = parsedContentType;
+                    }
```

Comment:
The WebAPI design for all of this was to make the header classes unable to accept invalid header values.

It was limiting in some scenarios where users had to interop with non-compliant other code - that's part of the reason why these are defined as strings on the http abstractions classes.

I think it's fine to add an encoding if it's valid... but we shouldn't mess with what the user set if it's invalid. That masks bugs.